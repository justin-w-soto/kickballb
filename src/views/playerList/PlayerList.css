a{
    color: white;
    list-style: none;
}