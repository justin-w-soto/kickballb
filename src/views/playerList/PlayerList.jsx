import React from 'react'
import { useEffect, useState } from "react"
import { Link } from "react-router-dom"
import { deletePlayerById, getPlayers } from "../../services/players"


export const PlayerList = () => {
    const [players, setPlayers] = useState([])
    const [loading, setLoading] =useState(true)

    const loadPlayer = async () => {
        const res = await getPlayers()
        setPlayers(res)
        setLoading(false)
    }

    useEffect(() => {loadPlayer()},[])

    const handleDelete = async ({id, name}) => {
        const shouldDelete = window.confirm(`Are you sure you want to delete ${name} from the roster?`)

        if(shouldDelete) 
        await deletePlayerById(id)
        await loadPlayer()
    }

    if (loading) return <p>Loading Players...</p>

    return (
        <div className="Playerlist">
              <h1>Players</h1>
              <Link to='/players/new' className='Toplink'>Add a New Player</Link>
              <ul>
                {players.map((player) => (
                    <li key={player.id}>

                    <Link className='link' to={`/players/${player.id}`}>
                        <li>{player.name}</li>
                    </Link>

                    <Link className='link' to={`players/${player.id}/update`}>
                        <button>Edit</button>
                    </Link>
                 
                    <button type="button" onClick={() => handleDelete({ id: player.id, name: player.name })}>Delete</button>
                </li>
                ))}
                </ul>
        </div>
    )
}
