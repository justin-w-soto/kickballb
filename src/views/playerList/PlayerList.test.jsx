import { render } from '@testing-library/react'
import { MemoryRouter, Route } from 'react-router-dom'
import { PlayerList } from './PlayerList';

it('Should render a list of players to the screen', async () => {
    const {container} = render(
        <MemoryRouter>
            <Route path="/players">
                <PlayerList />
            </Route>
        </MemoryRouter>
    )
    // screen.getByText('Players')
    expect(container).toMatchSnapshot()
})