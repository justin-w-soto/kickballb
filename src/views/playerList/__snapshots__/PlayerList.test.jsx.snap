// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a list of players to the screen 1`] = `<div />`;
